src/aggregation/metric/top_hits.rs

```diff
@@ -344,7 +366,7 @@ impl PartialEq for ComparableDocFeature {
 impl Eq for ComparableDocFeature {}
 
 #[derive(Clone, Serialize, Deserialize, Debug)]
-struct ComparableDocFeatures(Vec<ComparableDocFeature>, FieldRetrivalResult);
+struct ComparableDocFeatures(Vec<ComparableDocFeature>, IntermediateFieldRetrivalResult);
```

Comment:
Can we fix the terrible struct name and remove the 2 positional arguments and use a regular struct?

Reply:
Yes, I did also do a pass on all the names and some comments